tensorboard/plugins/image/images_plugin.py

```diff
@@ -353,12 +412,16 @@ def _get_individual_image(self, run, tag, index, sample):
     @wrappers.Request.application
     def _serve_individual_image(self, request):
         """Serves an individual image."""
-        run = request.args.get("run")
-        tag = request.args.get("tag")
-        index = int(request.args.get("index", "0"))
-        sample = int(request.args.get("sample", "0"))
         try:
-            data = self._get_individual_image(run, tag, index, sample)
+            if self._data_provider:
+                blob_key = request.args["blob_key"]
+                data = self._data_provider.read_blob(blob_key)
```

Comment:
If `read_blob` can raise NotFoundError, should we add that to the `except`?

Reply:
No need to; `errors.PublicError`s (including `NotFoundError`) can
propagate out of plugin code and are [handled by TensorBoard core][1],
so this will gracefully turn into a human-readable 404.

[1]: https://github.com/tensorflow/tensorboard/blob/f14006003b93d299d9713395ee4a4fef824b546e/tensorboard/backend/application.py#L181-L195

Reply:
Ah, I didn't know that!  That's convenient